### DISCUSSION TITLE
Pass arguments to cvlc fails

### DISCUSSION THREAD

**Discussion by kdeiss** [streamlink/streamlink#3965] [Q&A] [answered]
I'am using tvheadend and try to build additional channels with streamlink.

The way to do that is quite simple, because tvheadend has the feature to take input via pipe.
If I do this with the internal ffmpeg all is fine (e.g. pluto/eartcam) 

Sample m3u list looking like this:

`#EXTINF:-1,EC Amsterdam (EarthCam)
pipe:///home/hts/sbin/earthcamwrap https://www.earthcam.com/world/netherlands/amsterdam/?cam=amsterdam`

This eartcamwrap calls basically streamlink. Streamlink is piping the output, tvheadend accept that stream, all ok.

`/home/hts/.local/bin/streamlink $URL "best" --loglevel trace -O 2>>$SESSIONLOG`

If I do this with other streams e.g. youtube this is not working. I suppose the stream has to be remuxed.
So I try to call vlc, but not to play that stream - instead pipe the stream (--sout #std{mux=ts,access=file,dst=-).

When I do this with streamlink I receive an error. The call is as follows:

`/home/hts/.local/bin/streamlink --loglevel trace -v -p /usr/bin/cvlc -a "--sout #std{mux=ts,access=file,dst=-}" $URL "best"  2>>$SESSIONLOG`

Is there a trick to pass this kind of argument?

`Traceback (most recent call last):
  File "/home/hts/.local/bin/streamlink", line 11, in <module>
    load_entry_point('streamlink==2.3.0+28.g05bdf99', 'console_scripts', 'streamlink')()
  File "/root/.local/lib/python3.6/site-packages/streamlink_cli/main.py", line 1068, in main
    handle_url()
  File "/root/.local/lib/python3.6/site-packages/streamlink_cli/main.py", line 583, in handle_url
    handle_stream(plugin, streams, stream_name)
  File "/root/.local/lib/python3.6/site-packages/streamlink_cli/main.py", line 450, in handle_stream
    success = output_stream(plugin, stream)
  File "/root/.local/lib/python3.6/site-packages/streamlink_cli/main.py", line 314, in output_stream
    output.open()
  File "/root/.local/lib/python3.6/site-packages/streamlink_cli/output.py", line 24, in open
    self._open()
  File "/root/.local/lib/python3.6/site-packages/streamlink_cli/output.py", line 235, in _open
    self._open_subprocess()
  File "/root/.local/lib/python3.6/site-packages/streamlink_cli/output.py", line 257, in _open_subprocess
    args = self._create_arguments()
  File "/root/.local/lib/python3.6/site-packages/streamlink_cli/output.py", line 218, in _create_arguments
    args = self.args.format(**{PLAYER_ARGS_INPUT_DEFAULT: filename, PLAYER_ARGS_INPUT_FALLBACK: filename})
KeyError: 'mux=ts,access=file,dst=-'`

**Comment by bastimeyer** [comment, marked as answer]
> `-a "--sout #std{mux=ts,access=file,dst=-}"`
> `KeyError: 'mux=ts,access=file,dst=-'`

You need to escape curly braces in the `--player-args` / `-a` parameter value.

https://streamlink.github.io/cli.html#cmdoption-player-args

> The value can contain formatting variables surrounded by curly braces, { and }. If you need to include a brace character, it can be escaped by doubling, e.g. {{ and }}.

**Comment by kdeiss** [comment]
Thank you, I figured it already out. In this case a working solution was:

`/home/hts/.local/bin/streamlink --loglevel trace -v -p /usr/bin/cvlc -a "--sout \"#std{{mux=ts,access=file,dst=-}}\"" $URL "best"  2>>$SESSIONLOG`

The mentioned method is working nicely for earthcam. There moste streams are rtmp, piped this kind of stream to vlc is working. 

Same method with YouTube streams is not working. Tvheadend is creating a channel, but the quality is terrible, (1frame per minute or so). So I have to make another investigation whats wrong with YouTube streams ....



thanks

Klaus
